src/main/java/net/minecraftforge/common/extensions/IForgePackResources.java

```diff
@@ -5,10 +5,16 @@
 
 package net.minecraftforge.common.extensions;
 
+import net.minecraft.server.packs.PackType;
+
 public interface IForgePackResources
 {
     default boolean isHidden()
     {
         return false;
     }
+
+    default void initForNamespace(final String nameSpace) {}
+
+    default void init(final PackType packType) {}
```

Comment:
Could these two default to the behaviour in VanillaPackResources above - to avoid those extra patches there?

Reply:
No, since the cache manager is needed to init a given namespace properly, the first one is not changeable. The second one is debatable, be realistically only the vanilla pack has vanilla resources, and as such it makes no sense to have it default to that.